security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/action/StaticVarCompensatorActionDeserializer.java

```diff
@@ -0,0 +1,94 @@
+ */
+package com.powsybl.security.json.action;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.powsybl.commons.json.JsonUtil;
+import com.powsybl.iidm.network.StaticVarCompensator;
+import com.powsybl.security.action.StaticVarCompensatorAction;
+import com.powsybl.security.action.StaticVarCompensatorActionBuilder;
+
+import java.io.IOException;
+
+/**
+ * @author Etienne Lesot <etienne.lesot at rte-france.com>
+ */
+public class StaticVarCompensatorActionDeserializer extends StdDeserializer<StaticVarCompensatorAction> {
+
+    public StaticVarCompensatorActionDeserializer() {
+        super(StaticVarCompensatorAction.class);
+    }
+
+    private static class ParsingContext {
+        String id;
+        String staticVarCompensatorId;
+        String regulationMode;
+        String regulatingTerminal;
+        Double voltageSetPoint;
+        Double reactiveSetPoint;
+    }
+
+    @Override
+    public StaticVarCompensatorAction deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        ParsingContext context = new ParsingContext();
+        JsonUtil.parsePolymorphicObject(jsonParser, name -> {
+            switch (name) {
+                case "type":
+                    if (!StaticVarCompensatorAction.NAME.equals(jsonParser.nextTextValue())) {
+                        throw JsonMappingException.from(jsonParser, "Expected type " + StaticVarCompensatorAction.NAME);
+                    }
+                    return true;
+                case "id":
+                    context.id = jsonParser.nextTextValue();
+                    return true;
+                case "staticVarCompensatorId":
+                    context.staticVarCompensatorId = jsonParser.nextTextValue();
+                    return true;
+                case "regulationMode":
+                    context.regulationMode = jsonParser.nextTextValue();
+                    return true;
+                case "regulatingTerminal":
+                    context.regulatingTerminal = jsonParser.nextTextValue();
+                    return true;
+                case "voltageSetPoint":
+                    jsonParser.nextToken();
+                    context.voltageSetPoint = jsonParser.getValueAsDouble();
+                    return true;
+                case "reactiveSetPoint":
+                    jsonParser.nextToken();
+                    context.reactiveSetPoint = jsonParser.getValueAsDouble();
+                    return true;
+                default:
+                    return false;
+            }
+        });
+        StaticVarCompensator.RegulationMode mode = null;
+        if (context.regulationMode != null) {
+            mode = StaticVarCompensator.RegulationMode.valueOf(context.regulationMode);
+        }
```

Comment:
You can get the `StaticVarCompensator.RegulationMode` and use it in the builder in one step:
```java
if (context.regulationMode != null) {
    builder.withRegulationMode(StaticVarCompensator.RegulationMode.valueOf(context.regulationMode));
}
```

It will be easier to read and will remove one condition.